products/service.go

```diff
@@ -0,0 +1,121 @@
+package products
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/imadu/e-commerce-api/db"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var dbName = os.Getenv("DB_NAME")
+
+var productCollection = db.Client.Database(dbName).Collection("products")
+
+var ctx, cancel = context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
I think you no longer need timeout in this case, because it would have already canceled before any of the function gets called